fe/fe-core/src/main/java/com/starrocks/catalog/ScalarType.java

```diff
@@ -152,61 +153,67 @@ public static ScalarType createType(PrimitiveType type) {
         }
     }
 
-    public static ScalarType createType(String type) {
-        switch (type) {
-            case "INVALID_TYPE":
-                return INVALID;
-            case "NULL_TYPE":
-                return NULL;
-            case "BOOLEAN":
-                return BOOLEAN;
-            case "SMALLINT":
-                return SMALLINT;
-            case "TINYINT":
-                return TINYINT;
-            case "INT":
-                return INT;
-            case "BIGINT":
-                return BIGINT;
-            case "FLOAT":
-                return FLOAT;
-            case "DOUBLE":
-                return DOUBLE;
-            case "CHAR":
-                return CHAR;
-            case "VARCHAR":
-                return createVarcharType();
-            case "HLL":
-                return createHllType();
-            case "BITMAP":
-                return BITMAP;
-            case "PERCENTILE":
-                return PERCENTILE;
-            case "DATE":
-                return DATE;
-            case "DATETIME":
-                return DATETIME;
-            case "TIME":
-                return TIME;
-            case "DECIMAL":
-                return createDecimalV2Type();
-            case "DECIMALV2":
-                return createDecimalV2Type();
-            case "LARGEINT":
-                return LARGEINT;
-            case "DECIMAL32":
-                return DECIMAL32;
-            case "DECIMAL64":
-                return DECIMAL64;
-            case "DECIMAL128":
-                return DECIMAL128;
-            case "JSON":
-                return JSON;
-            default:
-                LOG.warn("type={}", type);
-                Preconditions.checkState(false);
-                return NULL;
+    public static Type createDecimal(StarRocksParser.DecimalTypeContext type) {
+        if (type.DECIMAL() != null) {
+            return ScalarType.createUnifiedDecimalType(10, 0);
+        } else if (type.DECIMALV2() != null) {
+            return ScalarType.createDecimalV2Type();
+        } else if (type.DECIMAL32() != null) {
+            return ScalarType.createDecimalV3Type(PrimitiveType.DECIMAL32);
+        } else if (type.DECIMAL64() != null) {
+            return ScalarType.createDecimalV3Type(PrimitiveType.DECIMAL64);
+        } else if (type.DECIMAL128() != null) {
+            return ScalarType.createDecimalV3Type(PrimitiveType.DECIMAL128);
+        } else {
+            throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
+        }
+    }
+
+    /**
+     * Create type from parser
+     */
+    public static Type createType(StarRocksParser.TypeContext type) {
```

Comment:
StarRocksParser.TypeContext itself is Antlr's internal data structure, which has nothing to do with the outside. This is a data structure that should be encapsulated inside AstBuilder. And now bringing it to ScalarType.java, there is nowhere to reuse this code, by spilling the types used inside Parser into other modules of the code, I don't think it's a good design

Reply:
ok, we could reverse this dependency, let the parser depends on `Type.java`. Same as before, most code could be reused.

Reply:
I totally agree